cmd/kube-proxy/app/server_others.go

```diff
@@ -92,6 +79,10 @@ func newProxyServer(
 	config *proxyconfigapi.KubeProxyConfiguration,
 	master string) (*ProxyServer, error) {
 
+	if config == nil {
```

Comment:
I think that confirms my bias towards keeping the extra validations, this could have detected the problem IIUIC the failures

Reply:
Just the opposite; kubeadm CI was starting kube-proxy in an invalid way for who knows who long, and no one ever noticed because it just returned an error message that no one paid attention to

Reply:
(actually, neither; that's not the line it's crashing on)

Reply:
I was trying to get a quick win, but most probably a red herring

Reply:
we don't know the reasons yet AFAIK, that was me trying to find an explanation